airbyte-integrations/connectors/source-mailchimp/unit_tests/test_streams.py

```diff
@@ -0,0 +1,127 @@
+from http import HTTPStatus
+from unittest.mock import MagicMock
+
+import json
+import pytest
+import responses
+from source_mailchimp.streams import (
+    Lists,
+    Campaigns,
+    EmailActivity
+)
+
+from utils import read_full_refresh, read_incremental
+
+
+@pytest.mark.parametrize(
+    ("http_status", "response_text", "expected_backoff_time"),
+    [
+        (HTTPStatus.BAD_GATEWAY, "", None),
+    ],
+)
+def test_backoff_time(auth, http_status, response_text, expected_backoff_time):
+    response_mock = MagicMock()
+    response_mock.status_code = http_status
+    response_mock.text = response_text
+    args = {"authenticator": auth}
+    stream = Lists(**args)
+    assert stream.backoff_time(response_mock) == expected_backoff_time
+
+
+@pytest.mark.parametrize(
+    "stream, endpoint",
+    [
+        (Lists, "lists"),
+        (Campaigns, "campaigns"),
+    ]
+)
+def test_stream_read(requests_mock, auth, stream, endpoint):
+    args = {"authenticator": auth}
+    stream = stream(**args)
+    responses = [
+        {
+            "json": {
+                stream.data_field: [
+                    {
+                        "id": "test_id"
+                    }
+                ],
+            }
+        }
+    ]
+    stream_url = stream.url_base + endpoint
+    requests_mock.register_uri("GET", stream_url, responses)
+    records = read_full_refresh(stream)
+
+    assert records
+
+
+def test_next_page_token(auth):
+    args = {"authenticator": auth}
+    stream = Lists(**args)
+    inputs = {"response": MagicMock()}
+    expected_token = None
+    assert stream.next_page_token(**inputs) == expected_token
+
+    resp = json.loads(json.dumps({"lists": [{"id": i} for i in range(101)]}))
```

Comment:
what's the need to do json.dumps and json.loads?

Reply:
remove it